core/arch/arm/arm.mk

```diff
@@ -119,16 +119,28 @@ endif
 core-platform-cflags += $(platform-cflags-optimization)
 core-platform-cflags += $(platform-cflags-generic)
 core-platform-cflags += $(platform-cflags-debug-info)
+ifeq ($(CFG_TEE_CORE_PIE),y)
+core-platform-cflags += -fpic
+endif
 
 core-platform-aflags += $(platform-aflags-generic)
 core-platform-aflags += $(platform-aflags-debug-info)
+ifeq ($(CFG_TEE_CORE_PIE),y)
+core-platform-aflags += -fpic
+endif
 
 ifeq ($(CFG_ARM64_core),y)
 arch-bits-core := 64
 core-platform-cppflags += $(arm64-platform-cppflags)
 core-platform-cflags += $(arm64-platform-cflags)
 core-platform-cflags += $(arm64-platform-cflags-generic)
 core-platform-cflags += $(arm64-platform-cflags-no-hard-float)
+ifeq ($(CFG_TEE_CORE_PIE),y)
+# "hidden" visibility to prevent the compiler from emitting
+# R_AARCH64_ABS64 relocations
+core-platform-cflags += -fvisibility=hidden
```

Comment:
Do this flag come with any cost? Any reason why we aren't always using it?
Why are the  R_AARCH64_ABS64 relocations emitted?

Reply:
> Do this flag come with any cost?

I don't think so, on the contrary it avoids a relocation.

> Any reason why we aren't always using it?

Well, no ;-) We could very well use it unconditionally I think.

> Why are the R_AARCH64_ABS64 relocations emitted?

As far as I can tell, they are just references to global symbols. When the compiler/linker generates a PIE, it assumes the global symbols could need to be resolved dynamically. So an entry is added to the symbol table and a relocation is emitted.